library/ssl_tls.c

```diff
@@ -1345,14 +1425,6 @@ static int ssl_encrypt_buf( mbedtls_ssl_context *ssl )
     MBEDTLS_SSL_DEBUG_BUF( 4, "before encrypt: output payload",
                       ssl->out_msg, ssl->out_msglen );
 
-    if( ssl->out_msglen > MBEDTLS_SSL_OUT_CONTENT_LEN )
```

Comment:
You seem to be moving this check from `ssl_encrypt_buf()` where it applies to all records, to `ssl_write_handshake_msg()` where it now only applies to handshake messages. Can you justify that the check was redundant for all non-handshake messages (that is, unless I'm mistaken, alerts and application data)? Off the top of my head I'm not sure I can. If not, please insert back this check here (and leave the new one in `ssl_write_handshake_msg()` in addition).

Reply:
I think you convinced me that there are enough other checks to make this one redundant, and thanks for pointing out if had become incorrect, I missed that. So, I'm not requesting any change here.